lib/winapi/sync.c

```diff
@@ -1,11 +1,145 @@
 #include <synchapi.h>
 #include <assert.h>
+#include <stdatomic.h>
 #include <stdbool.h>
 #include <processthreadsapi.h>
 #include <winbase.h>
 #include <winerror.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+static BOOL WINAPI condvar_init (PINIT_ONCE InitOnce, PVOID Parameter, PVOID *Context)
+{
+    PCONDITION_VARIABLE ConditionVariable = Parameter;
+    NTSTATUS status;
+    status = NtCreateEvent(&ConditionVariable->eventHandles[0], NULL, SynchronizationEvent, FALSE);
+    assert(NT_SUCCESS(status));
+    status = NtCreateEvent(&ConditionVariable->eventHandles[1], NULL, NotificationEvent, FALSE);
+    assert(NT_SUCCESS(status));
+    return true;
+}
+
+VOID InitializeConditionVariable (PCONDITION_VARIABLE ConditionVariable)
+{
+    ConditionVariable->initOnce.Ptr = 0;
+    ConditionVariable->eventHandles[0] = INVALID_HANDLE_VALUE;
+    ConditionVariable->eventHandles[1] = INVALID_HANDLE_VALUE;
+    atomic_store(&ConditionVariable->waitCount, 0);
```

Comment:
Wouldn't the `_Atomic` keyword take care of this? Why do we still need `atomic_store`?

Reply:
This was resolved by using builtins over `_Atomic` keyword.